Add a warning in the RMD report when some rows are lost after merging Diff and Fatures tables (#354)
nf-core/differentialabundance: assets/differentialabundance_report.Rmd

        qval_column = params$differential_qval_column
    )

    # If fold changes are not logged already, log them
    if (!params$differential_foldchanges_logged) {
        diff[[params$differential_fc_column]] <- log2(diff[[params$differential_fc_column]])
    }

    # Annotate differential tables if possible
    if (!is.null(params$features)) {

        # Merge tables

>         diff_features <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column)

Reviewer: The warnings are great. But I am wondering what would be the consequence of
```suggestion
        diff_features <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column, all.x = FALSE, all.y = TRUE)
```
would that lead to a complete plot (all features present) but with NA for (some) annotations? If so, that might be a nice solution as well.

Author: @d4straub As I can see it, the volcano plots won''t be affected, but I think this should be reported anyway (number of missing rows or number of features with NAs). We might take care of the new plots recently added below too (still in dev). I'm not sure if they can handle tables with NAs or without these columns at all @atrigila

        # Get number of rows before and after merging
        length_diff <- as.numeric(nrow(diff))
        length_features <- as.numeric(nrow(diff_features))

        # Compare numbers and report

>         if (length_diff != length_features) {

Reviewer 2: Not actually sure this is the right comparison. Maybe it would be better to check the dimensions of diff before and after the merge? Maybe also distinguish the possibility of there being MORE rows after the merge.

Author: @pinin4fjords I like the idea of adding the possibility of more rows, it would mean duplications, right? Working on that! But I would still compare only the number of rows, the dimensions (specifically the number of columns) will change based on the input parameters

Author: @pinin4fjords I agree that this should be addressed really upstream (as mentioned on the PR description), but the rmd code should contain the error handling as well, in case anyone runs it alone. What do you think?

Reviewer 2: Yeah, fair enough.

Author: @pinin4fjords Code updated, I changed a little bit the conditions to be more secure. It was true dimensions alone were not sufficient

            warnings <- c(warnings,
                paste0(

>                     '<p style="color:#DAA520;"><strong>WARNING:</strong> Some rows from the differential expressed table (', basename(diff_file), ') were lost after merging with features table (', basename(params$features), ').\n',

Reviewer 2: ```suggestion
                    '<p style="color:#DAA520;"><strong>WARNING:</strong> Some rows from the differential expressed table (', basename(diff_file), ') were absent from the features table (', basename(params$features), ') and lost on merge.\n',
```
